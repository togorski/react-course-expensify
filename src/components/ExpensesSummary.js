import React from "react"
import numeral from "numeral"
import { connect } from "react-redux"
import selectExpenses  from "../selectors/expenses"
import getTotalExpenses from "../selectors/expenses-total"

export const ExpensesSummary = (props) => (
    <div>
        <p>
            Viewing {props.expenseCount} expenses totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
        </p>
    </div>
)

const mapStateToProps = (state) => ({
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: getTotalExpenses(selectExpenses(state.expenses, state.filters))
})

export default connect(mapStateToProps)(ExpensesSummary)