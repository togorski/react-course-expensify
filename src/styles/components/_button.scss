.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    line-height: 1;
    font-weight: 300;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #d80909;
}