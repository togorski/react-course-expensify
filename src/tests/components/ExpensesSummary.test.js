import React from "react"
import { shallow } from "enzyme"
import { ExpensesSummary } from "../../components/ExpensesSummary"

test("should properly render with 1 expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={456}/>)
    expect(wrapper).toMatchSnapshot()
})

test("should properly render with multiple expenses", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenesTotal={123}/>)
    expect(wrapper).toMatchSnapshot()
})