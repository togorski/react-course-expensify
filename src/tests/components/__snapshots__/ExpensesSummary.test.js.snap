// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $4.56
  </h1>
</div>
`;

exports[`should properly render with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;
